src/TorchSharp/NN/Module.cs

```diff
@@ -1237,22 +1237,41 @@ public HookRemover register_forward_pre_hook(TPreHook hook)
                     return new HookRemover(this, key);
                 }
 
+                public HookRemover register_forward_hook(Action<TModule> hook)
+                {
+                    var key = Guid.NewGuid().ToString();
```

Comment:
Should we check whether the key exists now? I know that Guids are almost impossible to duplicate. But that might happen when two guid are generated at the same time...

Reply:
It would pretty much have to happen at the same nano-second.

Reply:
Also, since these keys are local to the module, you'd have to be registering a hook from two different threads at just the exact same moment.

Reply:
If we want to allow multi-threaded construction of models, which I'm not convinced is necessary, then we should instead add locks around the register/remove code.